src/main/java/de/bonndan/nivio/model/ComponentDiff.java

```diff
@@ -0,0 +1,61 @@
+package de.bonndan.nivio.model;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+
+public class ComponentDiff {
+
+    /**
+     * Compares changes in two strings.
+     *
+     * @param a       first
+     * @param b       second
+     * @param key     label
+     * @param changes array of changes
+     */
+    public static void compareStrings(String a, String b, String key, List<String> changes) {
+        if (!org.apache.commons.lang3.StringUtils.equals(a, b)) {
+            changes.add(key + " changed to: " + b);
+        }
+    }
+
+    /**
+     * Compares changes in string representation of two objects
+     *
+     * @param a       first
+     * @param b       second
+     * @param key     label
+     * @param changes array of changes
+     */
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    public static void compareOptionals(Optional<Object> a, Optional<Object> b, String key, List<String> changes) {
+        if (a.isEmpty() && b.isEmpty()) {
+            return;
+        }
+        if (a.isPresent() && b.isEmpty() || a.isEmpty() && b.isPresent()) {
```

Comment:
`b.isPresent()` is always true when reached, because it has been checked by the preceding conditions. Still, I would leave it there for better reliability of the code. I would suggest suppressing the warning and add a small comment to it.

Reply:
That's why I left it there. Refactored to clearify a bit